ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/FileUtils.java

```diff
@@ -75,18 +78,45 @@ public static boolean fileExists(final Path filePath) {
                     throw ex;
                 }
                 LOG.warn(String.format("WARN: file '%s' has an error and will be ignored", path.getFileName()));
-                ignoredBuilder.add(new PutBulk.IgnoreFile(path, ex.toString()));
+                ignoredBuilder.add(new FileUtils.IgnoreFile(path, ex.toString()));
             }
         }
 
-        return new PutBulk.ObjectsToPut(objectsBuilder.build(), ignoredBuilder.build());
+        return new FileUtils.ObjectsToPut(objectsBuilder.build(), ignoredBuilder.build());
+    }
+
+    /**
+     * Perform platform-specific normalization of path names on a list of Paths
+     * used by bulk get
+     * @param pipedFiles
+     * @return normalized pathname mapped to filename
+     */
+    public static ImmutableMap<String, String> getNormalizedObjectNames(final ImmutableList<Path> pipedFiles) {
+        final ImmutableMap.Builder<String, String> map = ImmutableMap.builder();
+        for (final Path file : pipedFiles) {
+            map.put(normalizeObjectName(file.toString()), file.toString());
+        }
+        return map.build();
+    }
```

Comment:
Instead of a very similar function to below, could we take the ImmutableList<Path>, create ImmutableList<String>, and call getNormalizedObjectNames(ImmutableList<String>) below?